packages/engine/Source/Scene/Picking.js

```diff
@@ -1266,6 +1276,12 @@ const scratchSurfaceNormal = new Cartesian3();
 const scratchSurfaceRay = new Ray();
 const scratchCartographic = new Cartographic();
 
+/**
+ * @private
+ * @param {Scene} scene
```

Comment:
It's a pity that this and several other functions receive **the** most complex (and most stateful) object there is - namely, the `Scene` - when all that they need is the (small and state-free(!)) `ellipsoid`. The function should always only have received the ellipsoid to begin with.